test/unit/Cardano/Wallet/BinarySpec.hs

```diff
@@ -109,6 +121,36 @@ spec = do
                     \8e3e95c150364fd775d4bc663ae6a9e6"
             hash `shouldBe` hash'
 
+
+    let pkWit = PublicKeyWitness "Trust me"
+
+    describe "Encoding Tx Witness" $ do
+        it "(decode . encode) succeeds" $ do
+            cborRoundtrip
+                decodeTxWitness
+                encodeTxWitness
+                pkWit
+
+    describe "Encoding Signed Tx" $ do
+        let txs = Set.toList (transactions block2 <> transactions block3)
+        it "(encode . decode) = pure" $ do
+            cborRoundtrip
+                decodeSignedTx
+                encodeSignedTx
+                (txs !! 0, [pkWit])
+
+
+cborRoundtrip
+    :: (HasCallStack, Show a, Eq a)
+    => (forall s. CBOR.Decoder s a)
+    -> (a -> CBOR.Encoding)
+    -> a
+    -> Expectation
+cborRoundtrip decode encode a = do
+    let bytes = CBOR.toLazyByteString $ encode a
+    let a' = unsafeDeserialiseFromBytes decode bytes
+    a `shouldBe` a'
```

Comment:
Two thoughts:
1. `Arbitrary` for roundtrips?
2. Use `ToCBOR`/`FromCBOR` classes more? (_I say despite having expressed scepticism of classes perviously_)